sources/packages/api/src/services/restriction/student-restriction.service.ts

```diff
@@ -255,4 +261,56 @@ export class StudentRestrictionService extends RecordDataModelService<StudentRes
     });
     return !!(await query.getRawOne());
   }
+  /**
+   * Service checks if the student has an active requested restriction.
+   * @param studentId student id.
+   * @param restrictionCode restriction code.
+   * @returns Student restrictions.
+   */
+  async isRestrictionExistsForStudent(
+    studentId: number,
+    restrictionCode: string,
+  ): Promise<boolean> {
+    return !!(await this.repo
+      .createQueryBuilder("studentRestrictions")
+      .select("studentRestrictions.id")
+      .innerJoin("studentRestrictions.restriction", "restriction")
+      .innerJoin("studentRestrictions.student", "student")
+      .where("student.id = :studentId", { studentId })
+      .andWhere("studentRestrictions.isActive = true")
+      .andWhere("restriction.restrictionCode = :restrictionCode", {
+        restrictionCode,
+      })
```

Comment:
Please add a `limit(1)`.

Reply:
do we need to have limit(1) when using getOne()?

Reply:
`.getOne` does not change the SQL query, which means that, if the query returns 5 records all of them will be retrieved from DB and then the `.getOne` will just take the first. Does it make sense?

Reply:
Yes, I get it. Thanks